src/CodeGen_C.cpp

```diff
@@ -585,6 +585,15 @@ void CodeGen_C::compile(const LoweredFunc &f) {
         stream << ") HALIDE_FUNCTION_ATTRS {\n";
         indent += 1;
 
+        // Emit a local user_context we can pass in all cases, either
+        // aliasing __user_context or nullptr.
+        if (!is_header()) {
+            do_indent();
+            stream << "void * const _ucon = " 
+                   << (have_user_context ? "(void *) __user_context" : "nullptr")
```

Comment:
If I understand this correctly, in at least some circumstances, __user_context could be "const void *". Is this what the PR fixes? If so, can this be done with const_cast<>?

Ideally we'd make __user_context non-const, but it may depend on how the user declared the parameters I guess. Point being if we're casting away constness, the signature of the filter routine should not allow it to be const....